cql3/statements/alter_table_statement.cc

```diff
@@ -320,8 +240,7 @@ future<shared_ptr<cql_transport::event::schema_change>> alter_table_statement::a
         break;
 
     case alter_table_statement::type::alter:
-        assert(_column_changes.size() == 1);
-        invoke_column_change_fn(std::mem_fn(&alter_table_statement::alter_column));
+        throw exceptions::invalid_request_exception("Altering of types is not allowed");
```

Comment:
I saw your comment about not removing this syntax because "it's not a major version", but I don't understand the logic: This patch completely removes this "alter" feature. It's not deprecated, nor does it prints a warning - rather this feature is outright removed. So does it matter if the error message the user gets when trying to use this suddenly-gone feature is a "syntax error" or one of "altering of types is not allowed"?
The benefit of removing this syntax completely is that you could remove a few more lines of code of type::alter definition, the parser code, etc.